### ISSUE TITLE
Variables declared inside a for-loop initialization result in block-scoped-var

### ISSUE THREAD

**Issue by wsmd** [eslint/eslint#5474]
**What version of ESLint are you using?**

2.2.0 through Atom's Linter.

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**

Only extending "airbnb/legacy". I haven't specified any parsers.

**What did you do? Please include the actual source code causing the issue.**

In this for loop, `new` is being declared within the initialization param of the for loop.
```javascript
function(array) {
  for (var new = array.slice(0), i = new.length - 1; i > 0; i--) {
    // do some stuff
  }
  return new; // < ESLint error
}
```
**What did you expect to happen?**

The variable `new` is still accessible and, in fact, it should! Why is ESLint complaining about it? 

According to MDN:

```
for ([initialization]; [condition]; [final-expression])
   statement
```

> initialization [...] is used to initialize a counter variable. This expression may optionally **declare new variables** with the var keyword. These variables **are not local to the loop**, i.e. **they are in the same scope the for loop is in.**

>

**What actually happened? Please include the actual, raw output from ESLint.**

`[ERROR] (block-scoped-var) 'new' used outside of binding context.`

**Comment by platinumazure** [comment]
My assumption is that the init statement of a for loop does not open a new block scope (its body does) and therefore this seems like a bug. I might have the spec wrong, though.

**Comment by mysticatea** [comment]
I believe this is intentional behavior.
`var` is not block-scoped, but `block-scoped-var` rule handles `var` as like block-scoped.

If you don't want to handle `var` as block-scoped, you can turn the rule off.

**Comment by michaelficarra** [comment]
Yeah, this is not a bug.

**Comment by ilyavolodin** [comment]
@wsmd This rule has been created to allow people coming from languages with block scopes to continue to use the same style. It ignores the fact that JavaScript has hoisting and pretends like everything is block scoped.
Closing, because this rule behaves as intended.
